src/vivarium/framework/engine.py

```diff
@@ -35,107 +33,139 @@
 from .plugins import PluginManager
 from .population import PopulationInterface
 from .randomness import RandomnessInterface
-from .results_writer import get_results_writer
 from .values import ValuesInterface
 from .time import TimeInterface
-from .lifecycle import LifeCycleInterface
+from .lifecycle import LifeCycleInterface, LifeCycleError
 
 _log = logging.getLogger(__name__)
 
 
 class SimulationContext:
-    """context"""
-    def __init__(self, configuration, components, plugin_manager=None):
-        plugin_manager = plugin_manager if plugin_manager else PluginManager()
 
-        self.configuration = configuration
+    def __init__(self, model_specification: Union[str, Path, ConfigTree] = None,
+                 components: Union[List, Dict[str, str], ConfigTree] = None,
+                 configuration: Union[Dict, ConfigTree] = None,
+                 plugin_configuration: Union[Dict[str, str], ConfigTree] = None):
+        # Bootstrap phase: Parse arguments, make private managers
+        component_configuration = components if isinstance(components, (dict, ConfigTree)) else None
```

Comment:
**Look here 2nd**

The only extra bit we have to do on the parsing is to potentially handle an incoming list of initialized components.